server/tracedb/azureappinsights.go

```diff
@@ -115,70 +115,181 @@ func (db *azureAppInsightsDB) GetTraceByID(ctx context.Context, traceID string)
 	return parseAzureAppInsightsTrace(traceID, table)
 }
 
-type columnIndex map[string]int
+type spanTable struct {
+	rows []spanRow
+}
+
+func (st *spanTable) Spans() []spanRow {
+	output := make([]spanRow, 0)
+	for _, row := range st.rows {
+		if row.Type() != "trace" {
+			output = append(output, row)
+		}
+	}
+
+	return output
+}
+
+func (st *spanTable) Events() []spanRow {
+	output := make([]spanRow, 0)
+	for _, row := range st.rows {
+		if row.Type() == "trace" {
+			output = append(output, row)
+		}
+	}
+
+	return output
+}
+
+type spanRow struct {
+	values map[string]any
+}
+
+func (sr *spanRow) Get(name string) any {
+	return sr.values[name]
+}
+
+func (sr *spanRow) Type() string {
+	return sr.values["itemType"].(string)
+}
+
+func (sr *spanRow) ParentID() string {
+	return sr.values["operation_ParentId"].(string)
+}
+
+func (sr *spanRow) SpanID() string {
+	return sr.values["id"].(string)
+}
+
+func newSpanTable(table *azquery.Table) spanTable {
+	spanRows := make([]spanRow, 0, len(table.Rows))
+	for _, row := range table.Rows {
+		spanRows = append(spanRows, newSpanRow(row, table.Columns))
+	}
+
+	return spanTable{spanRows}
+}
+
+func newSpanRow(row azquery.Row, columns []*azquery.Column) spanRow {
+	values := make(map[string]any)
+	for i, column := range columns {
+		name := *column.Name
+		if value := row[i]; value != nil {
+			values[name] = value
+		}
+	}
+
+	return spanRow{values}
```

Comment:
The reason I created those structs here is that AppInsights returns a table with the following fields:

```go
struct {
  Columns []string
  Rows    []struct {
    Values []string
  }
}
```

To get a value from a row, you have to know which index of the array it is. For that, you need to lookup the `Columns` array: 

```go
spanID := row[32]`
// or
`spanID := row[columns["name"]]`
```

So, by converting these fields into a `map[string]any` was an alternative that would make it easier to use (and debug).